orte/mca/rml/ofi/rml_ofi_send.c

```diff
@@ -0,0 +1,651 @@
+            opal_list_append(&new_msg->pkt_list, &new_pkt->super);
+            opal_list_append(&orte_rml_ofi.recv_msg_queue_list, &new_msg->super);
+            
+        }
+    }
+	
+}
+
+
+static void send_msg(int fd, short args, void *cbdata)
+{
+    orte_rml_send_request_t *req = (orte_rml_send_request_t*)cbdata;
+	orte_process_name_t *peer = &(req->send.dst);
+	orte_rml_tag_t tag = req->send.tag;
+	char *dest_ep_name;
+	size_t dest_ep_namelen = 0;
+	int ret = OPAL_ERROR;
+    uint32_t  total_packets;
+	fi_addr_t dest_fi_addr;
+	orte_rml_send_t *snd;
+	orte_rml_ofi_request_t* ofi_send_req = OBJ_NEW( orte_rml_ofi_request_t );
+	uint8_t conduit_id = req->conduit_id;
+    orte_rml_ofi_send_pkt_t* ofi_msg_pkt;
+    size_t datalen_per_pkt, hdrsize, data_in_pkt;  // the length of data in per packet excluding the header size
+   
+ 	snd = OBJ_NEW(orte_rml_send_t);
+    snd->dst = *peer;
+    snd->origin = *ORTE_PROC_MY_NAME;
+    snd->tag = tag;
+	if (NULL != req->send.iov) {
+        snd->iov = req->send.iov;
+        snd->count = req->send.count;
+        snd->cbfunc.iov = req->send.cbfunc.iov;
+    } else {
+        snd->buffer = req->send.buffer;
+        snd->cbfunc.buffer = req->send.cbfunc.buffer;
+    }
+    snd->cbdata = req->send.cbdata;
+
+    opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                         "%s send_msg_transport to peer %s at tag %d",
+                         ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),
+                         ORTE_NAME_PRINT(peer), tag);
+
+	
+	/* get the peer address by doing modex_receive 	*/
+    opal_output_verbose(10, orte_rml_base_framework.framework_output,
+                         "%s calling OPAL_MODEX_RECV_STRING ", ORTE_NAME_PRINT(ORTE_PROC_MY_NAME) );
+    switch ( orte_rml_ofi.ofi_conduits[conduit_id].fabric_info->addr_format)
+    {
+       case  FI_SOCKADDR_IN :
+             OPAL_MODEX_RECV_STRING(ret, OPAL_RML_OFI_FI_SOCKADDR_IN, peer , (char **) &dest_ep_name, &dest_ep_namelen);
+             break;
+       case  FI_ADDR_PSMX :
+             OPAL_MODEX_RECV_STRING(ret, OPAL_RML_OFI_FI_ADDR_PSMX, peer , (char **) &dest_ep_name, &dest_ep_namelen);
+             break;
+    }
+    opal_output_verbose(10, orte_rml_base_framework.framework_output,
+                         "%s  Return value from OPAL_MODEX_RECV_STRING - %d, length returned - %d", 
+                         ORTE_NAME_PRINT(ORTE_PROC_MY_NAME), ret, dest_ep_namelen);
+    /*print the sockaddr - port and s_addr */
+	struct sockaddr_in* ep_sockaddr = (struct sockaddr_in*) dest_ep_name;
+	opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                       "%s obtained for peer %s port = 0x%x, InternetAddr = 0x%x  ",
+                        ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),ORTE_NAME_PRINT(peer),ep_sockaddr->sin_port,
+                        ep_sockaddr->sin_addr.s_addr);
```

Comment:
Actually, I'm confused by this -- it looks like this module is designed to handle SOCKADDR_IN and PSMX, but this `opal_output_verbose()` unconditionally outputs the address in SOCKADDR_IN format.  Shouldn't it have to output the correct format?  Or should `fi_av_straddr(3)` be used?